frontend/src/pages/UserEditPage.jsx

```diff
@@ -1,14 +1,299 @@
-import { useParams } from 'react-router';
+import { useEffect, useState, useRef } from 'react';
+import { useParams, useNavigate } from 'react-router';
+import { useUser } from '../hooks/useUser';
+import { useApi } from '../hooks/useApi';
+import Spinny from '../components/Spinny';
+
 const UserEditPage = () => {
-  const { id } = useParams(); // Get the user ID from the URL
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const { isLogged, isAdmin } = useUser();
+  const [formData, setFormData] = useState({
+    username: '',
+    name: '',
+    is_admin: false,
+  });
+  const [formErrors, setFormErrors] = useState({});
+  const [showConfirmModal, setShowConfirmModal] = useState(false);
+  const confirmModalRef = useRef(null);
+
+  const {
+    data: fetchedUserData,
+    error: fetchError,
+    isLoading: isFetching,
+    fetchFromApi: fetchUser,
+  } = useApi();
```

Comment:
```suggestion
  } = useApi({ loadingInitially: true });
```
Since page needs to load data first, it can start with `isLoading === true`.